File: Kirameku-backend/app/schemas/font.py
"""font 字体相关 Schema（不含二进制文件数据，file_data 仅通过 /file 接口流式下发）"""
from pydantic import BaseModel


class FontOut(BaseModel):
    id: int
    name: str
    family: str
    role: str = "cjk"
    weight: int = 400
    license_name: str = ""
    license_url: str = ""
    file_name: str = ""
    mime_type: str = "font/woff2"
    file_size: int = 0
    enabled: bool = True
    is_default: bool = False
    sort: int = 0


class FontCreate(BaseModel):
    name: str
    family: str
    role: str = "cjk"
    weight: int = 400
    license_name: str = ""
    license_url: str = ""
    is_default: bool = False
    sort: int = 0


class FontUpdate(BaseModel):
    name: str | None = None
    family: str | None = None
    role: str | None = None
    weight: int | None = None
    license_name: str | None = None
    license_url: str | None = None
    enabled: bool | None = None
    is_default: bool | None = None
    sort: int | None = None

File: Kirameku-backend/admin/src/api/font.ts
import { http } from "@/utils/http";

export type FontItem = {
  id: number;
  name: string;
  family: string;
  role: string; // serif / sans / cjk / script
  weight: number;
  license_name: string;
  license_url: string;
  file_name: string;
  mime_type: string;
  file_size: number;
  enabled: boolean;
  is_default: boolean;
  sort: number;
};

/** 获取启用的字体列表（公开接口，后台同样可复用） */
export const getFonts = () => {
  return http.request<FontItem[]>("get", "/api/fonts");
};

/** 获取全部字体（含禁用，管理用） */
export const getAllFonts = () => {
  return http.request<FontItem[]>("get", "/api/fonts/all");
};